### ISSUE TITLE
Make DNS tests run in parallel

### ISSUE THREAD

**Issue by tripleee** [Charcoal-SE/SmokeDetector#6708]
As suggested by @makyen in #6703, look into running the DNS tests in parallel rather than serially, to make the overhead of testing shorter.

There is some support for this in the `dnspython` library already, so it should hopefully be rather straightforward.

```sh
tripleee$ wc -l *_nses.yml 
     125 blacklisted_nses.yml
    1539 watched_nses.yml
    1664 total
```

We are running over 1,500 name server tests in the test suite already, and the number is only likely to grow.

**Comment by makyen** [comment]
This has been implemented for CI testing in https://github.com/Charcoal-SE/SmokeDetector/commit/bb3098d5f6522667aff4527909a721f0076ca6ae. We should also run DNS checks within post scanning in parallel, but that should probably be another Issue.
